client/state/test/initial-state.js

```diff
@@ -810,22 +807,14 @@ describe( 'loading stored state with dynamic reducers', () => {
 	// For example, `withKeyPrefix( 'A' )` serializes `{ x: 1, y: 2 }` into `{ 'A:x': 1, 'A:y': 2 }`
 	const withKeyPrefix = ( keyPrefix ) => {
 		const keyPrefixRe = new RegExp( `^${ keyPrefix }:` );
-		const reducer = ( state = {}, action ) => {
-			switch ( action.type ) {
-				case DESERIALIZE:
-					return mapKeys( state, ( value, key ) => key.replace( keyPrefixRe, '' ) );
-				case SERIALIZE:
-					return mapKeys( state, ( value, key ) => `${ keyPrefix }:${ key }` );
-				default:
-					return state;
-			}
-		};
-		reducer.hasCustomPersistence = true;
-		return reducer;
+		return withPersistence( ( state = {} ) => state, {
+			serialize: ( state ) => mapKeys( state, ( value, key ) => `${ keyPrefix }:${ key }` ),
```

Comment:
A nice follow-up could be to replace `mapKeys()` usage with `Object.entries().map()` or something.